### ISSUE TITLE
Break into multi-project

### ISSUE THREAD

**Issue by pires** [nats-io/nats-connector-framework#17]
### Problem
With one project to rule them all, we'll have:

1. Dependency hell, exclusions everywhere for famous libs, i.e. Guava, Netty.
2. Continuous-integration, and artifact versioning/publishing is harder
3. Possible merging issues and confusing commit log

### Possible solutions
* Multi-module Maven project, one module per connector. [Fixes 1 and improves 2]
* Multi-repo, one per connector. Still, one parent POM should be used. [Fixes all]

@mcqueary @ColinSullivan1 please discuss!

**Comment by ColinSullivan1** [comment]
@pires  - Thank you, we were planning on doing this.  I think the second approach (multi-repo) is a good idea.

**Comment by mcqueary** [comment]
I agree with multi-repo with parent POM.

**Comment by ColinSullivan1** [comment]
@pires, I've refactored a bit - NATS now supports a connector framework (this repository), which other projects can use to create new connectors.  Other projects can specify the maven coordinates provided to include the NATS connector framework, now found in the sonatype repository.  I've separated out the Redis connector as the [NATS Redis Publish/Subscribe Connector](https://github.com/nats-io/nats-connector-redis).  This avoids any github/project dependencies between the connector framework and implemented connectors.
